### PR TITLE
chore(beta-api): promote handlers to v2.1 API

### LINKED ISSUE #826: Promote `/api/beta/discovery` to `/api/v2.1`

### PR DESCRIPTION
**Pull request by andrewazores**
Fixes #826 #827 #828
Depends on cryostatio/cryostat-web#374

### CODE REVIEW

File: src/test/java/io/cryostat/net/web/http/api/v2/DiscoveryGetHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    @Nested
+    class Requests {
+        @Mock RequestParameters params;
+        EnvironmentNode expected;
+
+        @BeforeEach
+        void setup() throws Exception {}
+
+        @Test
+        void shouldRespondWithEnvironmentNode() throws Exception {}
```
[the remark this answers is not in the document]
Comment (andrewazores) [reply src/test/java/io/cryostat/net/web/http/api/v2/DiscoveryGetHandlerTest.java:131]: Ah, hah. I noticed that that handler was missing tests so I set up this skeleton and then forgot to actually fill it in. Thanks for noticing.

[the remark this answers is not in the document]
Comment (andrewazores) [reply src/test/java/io/cryostat/net/web/http/api/v2/DiscoveryGetHandlerTest.java:131]: > Also, I'm assuming the empty @BeforeEach setup() inside this class overrides the one on line 72?

It doesn't override, they'll both end up running. I think you can even have multiple `@BeforeEach`s at the same "level" and they'll run in order, but definitely when there's one within a `@Nested` inner class then they both get run before each `@Test`.

[the remark this answers is not in the document]
Comment (hareetd) [reply src/test/java/io/cryostat/net/web/http/api/v2/DiscoveryGetHandlerTest.java]: Make sense, thanks.

**Review by jan-law — APPROVED** [review]
LGTM

**Review by hareetd — APPROVED** [review]

**Pull request merged**